java/src/main/java/org/hyperledger/fabric/client/FileCheckpointer.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2022 IBM All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.hyperledger.fabric.client;
+
+import com.google.gson.Gson;
+
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+class CheckpointerState {
+    private long blockNumber;
+    private String transactionId;
+    CheckpointerState(final long blockNumber, final String transactionId) {
+        this.blockNumber = blockNumber;
+        this.transactionId = transactionId;
+    }
+
+    public String getTransactionId() {
+        return transactionId;
+    }
+
+    public long getBlockNumber() {
+        return blockNumber;
+    }
+}
+
+/**
+ * FileCheckpointer to store checkpointer state during file read write operations.
+ */
+public final class FileCheckpointer implements Checkpointer {
```

Comment:
You might want to steal the file locking behaviour from https://github.com/hyperledger/fabric-gateway-java/blob/main/src/main/java/org/hyperledger/fabric/gateway/impl/FileCheckpointer.java

Reply:
It might be helpful to include a `toString()` implementation (see the old SDKs checkpointer implementation)